File: backend/app/schemas/product.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class ProductDocumentCreate(BaseModel):
    doc_type: str
    file_name: str
    file_path: str
    file_size: Optional[int] = 0
    description: Optional[str] = None
    sort_order: Optional[int] = 0


class ProductDocumentResponse(BaseModel):
    id: str
    product_id: str
    doc_type: str
    file_name: str
    file_path: str
    file_size: int
    description: Optional[str]
    sort_order: int
    created_at: datetime

    model_config = {"from_attributes": True}

class CategoryCreate(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    parent_id: Optional[str] = None
    sort_order: Optional[int] = 0

class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    parent_id: Optional[str] = None
    sort_order: Optional[int] = None
    is_active: Optional[bool] = None

class CategoryResponse(BaseModel):
    id: str
    name: str
    slug: str
    description: Optional[str]
    parent_id: Optional[str]
    sort_order: int
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class CategoryTreeResponse(BaseModel):
    id: str
    name: str
    slug: str
    description: Optional[str]
    parent_id: Optional[str]
    sort_order: int
    is_active: bool
    created_at: datetime
    children: list["CategoryTreeResponse"] = []

    model_config = {"from_attributes": True}

class ProductCreate(BaseModel):
    category_id: str
    name: str
    slug: str
    subtitle: Optional[str] = None
    description: Optional[str] = None
    technical_params: Optional[str] = None
    application_scenarios: Optional[str] = None
    advantages: Optional[str] = None
    specifications: Optional[dict] = None
    specifications_text: Optional[str] = None
    density: Optional[str] = None
    strength: Optional[str] = None
    thermal_conductivity: Optional[str] = None
    unit_weight: Optional[str] = None
    fire_rating: Optional[str] = None
    image_url: Optional[str] = None
    meta_title: Optional[str] = None
    meta_description: Optional[str] = None
    sort_order: Optional[int] = 0

class ProductUpdate(BaseModel):
    category_id: Optional[str] = None
    name: Optional[str] = None
    slug: Optional[str] = None
    subtitle: Optional[str] = None
    description: Optional[str] = None
    technical_params: Optional[str] = None
    application_scenarios: Optional[str] = None
    advantages: Optional[str] = None
    specifications: Optional[dict] = None
    specifications_text: Optional[str] = None
    density: Optional[str] = None
    strength: Optional[str] = None
    thermal_conductivity: Optional[str] = None
    unit_weight: Optional[str] = None
    fire_rating: Optional[str] = None
    image_url: Optional[str] = None
    meta_title: Optional[str] = None
    meta_description: Optional[str] = None
    sort_order: Optional[int] = None
    is_active: Optional[bool] = None

class ProductResponse(BaseModel):
    id: str
    category_id: str
    name: str
    slug: str
    subtitle: Optional[str]
    description: Optional[str]
    technical_params: Optional[str]
    application_scenarios: Optional[str]
    advantages: Optional[str]
    specifications: Optional[dict]
    specifications_text: Optional[str]
    density: Optional[str]
    strength: Optional[str]
    thermal_conductivity: Optional[str]
    unit_weight: Optional[str]
    fire_rating: Optional[str]
    image_url: Optional[str]
    meta_title: Optional[str]
    meta_description: Optional[str]
    sort_order: int
    is_active: bool
    view_count: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/stores/product.ts
import { defineStore } from 'pinia'
import { useApi } from '~/composables/useApi'

interface ProductCategory {
  id: string
  name: string
  slug: string
  description: string | null
  parent_id: string | null
  sort_order: number
  is_active: boolean
  created_at: string
}

interface Product {
  id: string
  category_id: string
  name: string
  slug: string
  subtitle: string | null
  description: string | null
  technical_params: string | null
  application_scenarios: string | null
  advantages: string | null
  specifications: string | null
  density: string | null
  strength: string | null
  thermal_conductivity: string | null
  unit_weight: string | null
  image_url: string | null
  meta_title: string | null
  meta_description: string | null
  sort_order: number
  is_active: boolean
  view_count: number
  created_at: string
}

export const useProductStore = defineStore('product', {
  state: () => ({
    categories: [] as ProductCategory[],
    products: [] as Product[],
    currentProduct: null as Product | null,
    loading: false,
    error: null as string | null,
  }),
  getters: {
    activeProducts: (state) => state.products.filter((p) => p.is_active),
    categoryMap: (state) => {
      const map: Record<string, ProductCategory> = {}
      for (const cat of state.categories) {
        map[cat.id] = cat
      }
      return map
    },
  },
  actions: {
    async fetchCategories() {
      this.loading = true
      this.error = null
      try {
        const { request } = useApi()
        this.categories = await request<ProductCategory[]>('/products/categories')
      } catch (e: any) {
        this.error = e.message
      } finally {
        this.loading = false
      }
    },
    async fetchProducts(categoryId?: string) {
      this.loading = true
      this.error = null
      try {
        const { request } = useApi()
        const response = await request<{ items: Product[], total: number }>('/products', {
          params: categoryId ? { category_id: categoryId } : undefined,
        })
        this.products = response.items || []
      } catch (e: any) {
        this.error = e.message
      } finally {
        this.loading = false
      }
    },
    async fetchProductBySlug(slug: string) {
      this.loading = true
      this.error = null
      try {
        const { request } = useApi()
        this.currentProduct = await request<Product>(`/products/by-slug/${slug}`)
      } catch (e: any) {
        this.error = e.message
      } finally {
        this.loading = false
      }
    },
  },
})
